wrappers/rust/icicle-core/src/poseidon2/mod.rs

```diff
@@ -53,20 +64,41 @@ macro_rules! impl_poseidon2 {
 
             extern "C" {
                 #[link_name = concat!($field_prefix, "_create_poseidon2_hasher")]
-                fn create_poseidon2_hasher(t: u32, domain_tag: *const $field) -> HasherHandle;
+                fn create_poseidon2_hasher(t: u32, domain_tag: *const $field, input_size: u32) -> HasherHandle;
             }
 
             // Implement the `Poseidon2Hasher` trait for the given field configuration.
             impl Poseidon2Hasher<$field> for $field_cfg {
                 fn new(t: u32, domain_tag: Option<&$field>) -> Result<Hasher, eIcicleError> {
                     let handle: HasherHandle = unsafe {
-                        create_poseidon2_hasher(t, domain_tag.map_or(std::ptr::null(), |tag| tag as *const $field))
+                        create_poseidon2_hasher(
+                            t,
+                            domain_tag.map_or(std::ptr::null(), |tag| tag as *const $field),
+                            0,
+                        )
```

Comment:
Can we change this to call the `new_with_input_size` function internally with an input size of 0.